src/main/java/uk/gov/pay/api/clients/ConnectorClient.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.pay.api.clients;
+
+import org.slf4j.MDC;
+
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import java.util.Map;
+import java.util.Optional;
+
+public class ConnectorClient {
+
+    private final Client client;
+
+    @Inject
+    public ConnectorClient(Client client) {
+        this.client = client;
+    }
+
+    public Response get(String url) {
+        MultivaluedMap<String, Object> headers = getHeaders();
+        return client.target(url).request().headers(headers).accept(MediaType.APPLICATION_JSON).get();
+    }
+
+    private MultivaluedMap<String, Object> getHeaders() {
+        return Optional.ofNullable(MDC.get("x_request_id"))
+                    .filter(s -> !s.isBlank())
+                    .map(s -> new MultivaluedHashMap<String, Object>(Map.of("X-Request-Id", s)))
+                    .orElse(new MultivaluedHashMap<>());
```

Comment:
not saying change it, but is it really necessary to do the streaming stuff here? I guess it comes down to personal preference

Reply:
actually I'd prefer it with the streaming stuff as `return new MultivaluedHashMap<>(Map.of("X-Request-Id", MDC.get("x_request_id")));` will throw an exception if `MDC.get("x_request_id")` returns null.

Reply:
I more meant just doing a standard if statement checking whether the result is blank.

Reply:
I don't think we should get into personal preferences... it's works for the functionality we're trying to achieve.

Reply:
yeh sure